[2.2][Backport] Database Media Storage - Design Config Save functions to be Database Media Storage aware

gwharton opened the pull request:
### Original PR (*)
#21675 

### Description (*)
The functions that process the uploads of the Favicon, Header Image and Transactional Email Logo files do not process the file correctly when in Database Media Storage Mode. This PR fixes that.

### Fixed Issues (if relevant)
1. magento/magento2#21672: Database Media Storage - Design Config fails to save transactional email logo correctly

### Manual testing scenarios (*)
1. Setup Magento
2. Change media storage mode to database
3. Synchronize
4. Save Settings
5. Content->Configuration->Edit->Transactional Emails
6. Upload New Image (but don't save)
7. Verify temporary email logo exists in pub/media
8. Verify temporary email logo exists in database
9. Save Config
10. Verify email logo exists in pub/media
11. Verify email logo exists in database

Repeat for favicon and header logo, ensuring that after save, the file exists correctly in the database.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

magento-engcom-team commented:
Hi @gwharton. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.2-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

VasylShvorak commented:
On hold until [#21675](https://github.com/magento/magento2/pull/21675) would be merged

Commit:
When Saving design config, ensure that temporary file in database is renamed aswell as the local file in pub/media.

Initialise Database Helper from Object Manager if not passed by DI. Increases backwards compatability.

magento-engcom-team commented:
Hi @sidolov, thank you for the review.
[ENGCOM-4801](https://jira.corp.magento.com/browse/ENGCOM-4801) has been created to process this Pull Request

VasylShvorak commented:
:heavy_check_mark: QA passed
